src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.cs

```diff
@@ -67,13 +66,34 @@ private GCHandle(object? value, GCHandleType type)
         /// <returns>A new GC handle that protects the object.</returns>
         public static GCHandle Alloc(object? value, GCHandleType type) => new GCHandle(value, type);
 
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            IntPtr handle = _handle;
+
+            if ((nint)handle != 0)
+            {
+                InternalFree(GetHandleValue(handle));
+
+                _handle = IntPtr.Zero;
+            }
```

Comment:
It probably doesn't matter, but I'd keep the behavior of nulling out the field prior to freeing the value.
```suggestion
            IntPtr handle = _handle;
            if ((nint)handle != 0)
            {
                _handle = IntPtr.Zero;
                InternalFree(GetHandleValue(handle));
            }
```

Reply:
Sure thing, done in ec81ed110f3173edb87b5316e1bd9b99c111e725! 😄
I figured since this was explicitly not thread-safe it would be the same but I agree it's better this way.